manifests/charts/istiod-remote/values.yaml

```diff
@@ -1,455 +0,0 @@
-      readinessFailureThreshold: 4
-      # The initial delay for readiness probes in seconds.
-      readinessInitialDelaySeconds: 0
-      # The period between readiness probes.
-      readinessPeriodSeconds: 15
-      # Enables or disables a startup probe.
-      # For optimal startup times, changing this should be tied to the readiness probe values.
-      #
-      # If the probe is enabled, it is recommended to have delay=0s,period=15s,failureThreshold=4.
-      # This ensures the pod is marked ready immediately after the startup probe passes (which has a 1s poll interval),
-      # and doesn't spam the readiness endpoint too much
-      #
-      # If the probe is disabled, it is recommended to have delay=1s,period=2s,failureThreshold=30.
-      # This ensures the startup is reasonable fast (polling every 2s). 1s delay is used since the startup is not often ready instantly.
-      startupProbe:
-        enabled: true
-        failureThreshold: 600 # 10 minutes
-      # Resources for the sidecar.
-      resources:
-        requests:
-          cpu: 100m
-          memory: 128Mi
-        limits:
-          cpu: 2000m
-          memory: 1024Mi
-      # Default port for Pilot agent health checks. A value of 0 will disable health checking.
-      statusPort: 15020
-      # Specify which tracer to use. One of: zipkin, lightstep, datadog, stackdriver, none.
-      # If using stackdriver tracer outside GCP, set env GOOGLE_APPLICATION_CREDENTIALS to the GCP credential file.
-      tracer: "none"
-    proxy_init:
-      # Base name for the proxy_init container, used to configure iptables.
-      image: proxyv2
-      # Bypasses iptables idempotency handling, and attempts to apply iptables rules regardless of table state, which may cause unrecoverable failures.
-      # Do not use unless you need to work around an issue of the idempotency handling. This flag will be removed in future releases.
-      forceApplyIptables: false
-    # configure remote pilot and istiod service and endpoint
-    remotePilotAddress: ""
-    ##############################################################################################
-    # The following values are found in other charts. To effectively modify these values, make   #
-    # make sure they are consistent across your Istio helm charts                                #
-    ##############################################################################################
-
-    # The customized CA address to retrieve certificates for the pods in the cluster.
-    # CSR clients such as the Istio Agent and ingress gateways can use this to specify the CA endpoint.
-    # If not set explicitly, default to the Istio discovery address.
-    caAddress: ""
-    # Configure a remote cluster data plane controlled by an external istiod.
-    # When set to true, istiod is not deployed locally and only a subset of the other
-    # discovery charts are enabled.
-    externalIstiod: true
```

Comment:
@bleggett did you intentionally not move `externalIstiod: true` over to `profile-remote.yaml`? Without this, the primary-remote setup, as described [here](https://istio.io/latest/docs/setup/install/multicluster/primary-remote/), no longer works. Istiod shows the following error:
```
error	klog	error retrieving resource lock istio-system/istio-namespace-controller-election: configmaps "istio-namespace-controller-election" is forbidden: User "system:serviceaccount:istio-system:istio-reader-service-account" cannot get resource "configmaps" in API group "" in the namespace "istio-system"
```

This is because this privilege is only included in the reader clusterrole when `externalIstiod` is `true` - see [here](https://github.com/istio/istio/blob/b722511e45f69d8bd98de9980084db6a8bebf7a3/manifests/charts/istio-control/istio-discovery/templates/reader-clusterrole.yaml#L52-L62).

Reply:
Furthermore, when `externalIstiod` is `false`, the Service and Endpoints are now called `istiod-remote`, but the `istio-sidecar-injector` webhook points to the non-existent `istiod` service.

Reply:
Here's a (preliminary) [fix](https://github.com/istio/istio/pull/53542).

Reply:
> @bleggett did you intentionally not move `externalIstiod: true` over to `profile-remote.yaml`? Without this, the primary-remote setup, as described [here](https://istio.io/latest/docs/setup/install/multicluster/primary-remote/), no longer works. Istiod shows the following errors:
> 
Yes, see https://github.com/istio/istio.io/pull/15776 for a related docs fix.

`externalIstiod` is a bad flag - our tests use it in contradictory ways. It's pretty clearly stated in the docs and in the inline description of that flag that `externalistiod` should be set **on the cluster that exposes** the control plane.

But in fact, it is required on both ends, which is not how it's documented.

This PR mostly left `externalIstiod` as-is (that is, IMO, broken and misdocumented).

IMO the flag simply should be removed and happy too look at followups for that, but that was OOS of this PR.